doc/BUILD.rst

```diff
@@ -0,0 +1,722 @@
+
+* Default: ``false``
+* Options: ``true``, ``false``
+
+This option forces stricter checks against Taisei's codebase to ensure code
+health, treating all ``Warnings`` as ``Errors`` in the code.
+
+It's highly recommended to **enable** (i.e: ``true``) this whenever developing
+for the engine. Sometimes, it's overly-pedantic, but much of the time, it
+provides useful advice. (For example, it can detect potential null-pointer
+exceptions that may not be obvious to the human eye.)
+
+.. code:: sh
+
+   meson configure build/ -Dwerror=true
+
+Deprecation Warnings (``-Ddeprecation_warnings``)
+"""""""""""""""""""""""""""""""""""""""""""""""""
+
+* Default: ``(null)``
+* Options: ``error``, ``no-error``, ``ignore``
+
+Sets deprecation warnings to either hard-fail (``error``), print as warnings but
+not trigger full errors if ``-Dwerror=true`` (``no-error``), and otherwise
+ignore them (``ignore``).
+
+Generally, ``no-error`` is the recommended default when using ``-Dwerror=true``.
+
+.. code:: sh
+
+   meson configure build/ -Ddeprecation_warnings=no-error
+
+
+Stack Trace Debugging (``-Db_sanitize``)
+""""""""""""""""""""""""""""""""""""""""
+
+This is useful for debugging crashes in the game. It uses
+`AddressSanitizer <https://github.com/google/sanitizers/wiki/AddressSanitizer>`__:
+
+.. code:: sh
+
+   meson configure build/ -Db_sanitize=address,undefined
+
+Depending on your platform, you may need to specify the specific library binary
+to use to launch ASan appropriately. Using macOS as an example:
+
+.. code:: sh
+
+   export DYLD_INSERT_LIBRARIES=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/12.0.0/lib/darwin/libclang_rt.asan_osx_dynamic.dylib
+
+The ``../12.0.0/..`` in the path of ``DYLD_INSERT_LIBRARIES`` changes with each
+version of Xcode. If it fails to launch for you, ensure that the version number
+is correct by browsing to the parent directory of ``../clang``.
+
+Then, you can launch Taisei's binary from the command line (using macOS as an
+example):
+
+.. code:: sh
+
+   /path/to/Taisei.app/Contents/MacOS/Taisei
+
+Link-Time Optimizations (``-Db_lto``)
+"""""""""""""""""""""""""""""""""""""
+
+* Default: ``true``
+* Options: ``true``, ``false``
+
+Link-time optimizations (LTO) increase build times, but also increase
+performance. For quicker build times during development, you can disable it.
+For release builds, this should be keep ``true``.
+
+See: `Interprocedural Optimization <https://en.wikipedia.org/wiki/Interprocedural_optimization#WPO_and_LTO>`__
+
+.. code:: sh
+
+   meson configure build/ -Db_lto=false
+
+Binary Striping (``-Dstrip``)
+"""""""""""""""""""""""""""""
+
+* Default: ``true``
+* Options: ``true``, ``false``
+
+This option prevents stripping of the `taisei` binary, leading to faster build
+times and keeping debugging symbols in place. There is a theoretical performance
+hit with this option enabled, but it can help with building during development.
```

Comment:
I don't think this affects build performance at all. This applies to the binary on installation only. Stripping is also very very fast compared to every other build step. 

Enabling this absolutely does *not* help during development, quite the opposite since you usually want to have debugging symbols in.